Towny/src/main/java/com/palmergames/bukkit/towny/listeners/TownyCustomListener.java

```diff
@@ -363,4 +394,48 @@ private void attemptPlayerCacheReset(Player player, WorldCoord worldCoord) {
 			return;
 		Towny.getPlugin().resetCache(player);
 	}
+
+	/*
+	 * Watch for town and nation level increasing/decreasing and reassign permissions in case the players have level-requirement permissions. 
+	 */
+
+	@EventHandler
+	public void onTownLevelIncrease(TownLevelIncreaseEvent event) {
```

Comment:
These four events could be merged together into one single GovernmentLevelChangedEvent to reduce the amount of repeated code here, getters could also be added for the old level and new level.